python_modules/dagster/dagster/_core/definitions/declarative_scheduling/operators/boolean_operators.py

```diff
@@ -25,9 +25,9 @@ def description(self) -> str:
     def evaluate(self, context: SchedulingContext) -> SchedulingResult:
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 15]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Question: does the `Sequence` type in Python guarantee stable ordering? We might want to consider typing this as `List` which for sure guarantees that. We're relying on index for identity so just want to be sure.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21671?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
From what I can tell, the answer is yes -- I think in general if an object supports index operations (which all Sequences do), then ordering must be stable.